### PR TITLE
Binary compatibility

### PR DESCRIPTION
**Pull request by pepone**
This PR includes a few minor fixes to restore binary compatibility with 3.7.5, there is also a minor fix for InputStream compiler warnings when building with clang in C++20 mode

### CODE REVIEW

**Comment by externl** [comment]
O nmacOS when I set the compiler to C++ 17 mode (`-std=c++17`) I get:

```
include/Ice/Object.h:332:13: error: definition of implicit copy constructor for 'Object' is deprecated because it has a user-declared destructor [-Werror,-Wdeprecated-copy-dtor]
    virtual ~Object() {}
            ^
include/generated/Ice/Metrics.h:2484:5: note: in implicit copy constructor for 'Ice::Object' first required here
    Metrics(const Metrics&) = default;
    ^
src/Ice/generated/Metrics.cpp:1270:28: note: in defaulted copy constructor for 'IceMX::Metrics' first required here
    ::Ice::Object* p = new Metrics(*this);
```

File: cpp/src/IceStorm/Replica.h [hunk 1]
```diff
@@ -22,9 +22,9 @@ struct GroupNodeInfo
     // COMPILER FIX: Clang using libc++ requires to define operator=
     //
 #if defined(__clang__) && defined(_LIBCPP_VERSION)
-#ifdef ICE_CPP11_COMPILER
-    GroupNodeInfo(const GroupNodeInfo&) = default;
-#endif
+#   ifdef ICE_CPP11_COMPILER
+    GroupNodeInfo(const GroupNodeInfo&);
```
Comment (bernardnormier) [inline cpp/src/IceStorm/Replica.h:26]: Since you removed the `= default`, does it mean it should be `= delete`?

**Review by bernardnormier — APPROVED** [review]
Looks fine. Are the "new" virtual destructors present in 3.7.5?

**Comment by pepone** [comment]
> On macOS when I set the compiler to C++ 17 mode (-std=c++17) I get:

We can disable this with a pragma but there are other failures in different places, at least with `--std=c++20` that I tested, and removing this would break Windows binary compatibility

**Comment by pepone** [comment]
> Are the "new" virtual destructors present in 3.7.5?

Yes this restore the ones removed in

https://github.com/zeroc-ice/ice/commit/c095b7f9c123e1b07bf3844e7aaad9bb1a56d143

File: cpp/src/IceStorm/Replica.h (hunk 1 above)
Comment (pepone) [reply cpp/src/IceStorm/Replica.h:26]: I think `= default` is to force the generation of the copy constructor on C++17 mode or greater, we can probably keep this `=default` as this is internal, shouldn't be an issue for binary compatibility.

**Review by externl — APPROVED** [review]

**Pull request merged**